volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -279,6 +279,77 @@ def get_symbol_by_address(self, wanted_sym_address):
 
         return None
 
+    def _module_flags_taints_pre_4_10_rc1(self) -> str:
+        """Convert the module's taints value to a 1-1 character mapping.
+        Relies on statically defined taints mappings in the framework.
+
+        Returns:
+            The raw taints string.
+        """
+        taints_string = ""
+        for char, taint_flag in linux_constants.TAINT_FLAGS.items():
+            if taint_flag.module and self.taints_value & taint_flag.shift:
+                taints_string += char
+
+        return taints_string
+
+    def _module_flags_taints_post_4_10_rc1(self) -> str:
+        """Convert the module's taints value to a 1-1 character mapping.
+        Relies on kernel symbol embedded taints definitions.
+
+            struct taint_flag {
+                char c_true;		/* character printed when tainted */
+                char c_false;		/* character printed when not tainted */
+                bool module;		/* also show as a per-module taint flag */
+            };
+
+        Returns:
+            The raw taints string.
+        """
+        taints_string = ""
+        for i, taint_flag in enumerate(self.taint_flags_list):
+            c_true = chr(taint_flag.c_true)
+            c_false = chr(taint_flag.c_false)
+            if taint_flag.module and (self.taints_value & (1 << i)):
+                taints_string += c_true
+            elif taint_flag.module and c_false != " ":
+                taints_string += c_false
+
+        return taints_string
+
+    def get_taints_as_plain_string(self) -> str:
```

Comment:
These are externally visible additions to the API, which means a MINOR version number somewhere, needs to go up.  I suspect it may be the framework itself because I don't think anything further down is versioned?

Reply:
As discussed, we want to include this API in a more generic place, but keep these convenient self-contained functions. So right now I'm not sure of how many bumps are needed.